sima/motion/dftreg.py

```diff
@@ -0,0 +1,788 @@
+"""
+Motion correction of image sequences by 'efficient subpixel image registration
+by cross correlation'. A reference image is iteratively computed by aligning
+and averaging a subset of images/frames.
+2015 Lloyd Russell, Christoph Schmidt-Hieber
+
+*******************************************************************************
+Credit to Marius Pachitariu for concept of registering to aligned mean image.
+
+Parts of the code are based on:
+skimage.feature.register_translation, which is a port of MATLAB code by Manuel
+Guizar-Sicairos, Samuel T. Thurman, and James R. Fienup, "Efficient subpixel
+image registration algorithms," Optics Letters 33, 156-158 (2008).
+
+Relating to implementation of skimage.feature.register_translation:
+Copyright (C) 2011, the scikit-image team
+All rights reserved.
+THIS SOFTWARE IS PROVIDED BY THE AUTHOR ''AS IS'' AND ANY EXPRESS OR IMPLIED
+WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+OF SUCH DAMAGE.
+*******************************************************************************
+@author: llerussell
+"""
+
+from __future__ import absolute_import, division
+from builtins import map, range
+from functools import partial
+import multiprocessing
+import numpy as np
+from scipy.ndimage.interpolation import shift
+import time
+from . import motion
+
+
+class DiscreteFourier2D(motion.MotionEstimationStrategy):
+    """
+    Motion correction of image sequences by 'efficient subpixel image
+    registration by cross correlation'. A reference image is iteratively
+    computed by aligning and averaging a subset of images/frames.
+
+    Parameters
+    ----------
+    upsample_factor : int
```

Comment:
Our convention for optional arguments has been as follows (also, please try to make sure things are as clear as possible by following roughly the style below):

```
upsample_factor : int, optional
    The final pixel alignment has resolution of 1/upsample_factor. 
    If set to 1, only pixel level shifts are made, in which case 
    the algorithm runs faster and does not interpolate 
    between pixels. Default: 1.
```